BrainPortal/app/controllers/bourreaux_controller.rb

```diff
@@ -610,6 +610,29 @@ def rr_access_dp
 
   end
 
+  # API method to copy files from one DP to another via a bourreau
+  def file_copy #:nodoc:
```

Comment:
Rejected for security. This action has no validation or verifications. It will copy any files to anywhere. Which means that if we deployed this, a normal user could copy any file, including those that belongs to the admins or other users, to their own data provider.